azuki-core/src/main/kotlin/com/anaplan/engineering/azuki/core/system/Implementation.kt

```diff
@@ -16,12 +16,29 @@ interface Implementation<AF : ActionFactory, CF : CheckFactory, QF : QueryFactor
 
     fun createSystemFactory(systemDefaults: SD = implementationDefaults): SystemFactory<AF, CF, QF, AGF, SD>
 
+    val versionFilter: VersionFilter
+
     companion object {
         fun <AF : ActionFactory, CF : CheckFactory, QF : QueryFactory, AGF : ActionGeneratorFactory> locateImplementations(): List<Implementation<AF, CF, QF, AGF, *>> {
             val loader = ServiceLoader.load(Implementation::class.java)
             return loader.iterator().asSequence().filterIsInstance<Implementation<AF, CF, QF, AGF, *>>().toList()
         }
     }
+
+    interface VersionFilter {
+
+        /**
+         * Provides a means for an implementation to skip verification of scenario based on version it was introduced
```

Comment:
⛏️ I find this sentence a bit hard to parse. I had to read it three times to be sure of what it meant.